graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -280,7 +287,32 @@ def _check_for_cascading_type_suppression(
         raise CascadingSuppressionError("\n".join(error_message_components))
 
 
-def _ensure_no_unsupported_suppression(
+def _ensure_no_unsupported_operations(
+    schema_ast: DocumentNode, renamings: Mapping[str, Optional[str]], scalars: AbstractSet[str],
+) -> None:
+    """Check for unsupported renaming or suppression operations."""
+    _ensure_no_unsupported_scalar_operations(renamings, scalars)
+    _ensure_no_unsupported_suppressions(schema_ast, renamings)
+
+
+def _ensure_no_unsupported_scalar_operations(
+    renamings: Mapping[str, Optional[str]], scalars: AbstractSet[str],
+) -> None:
+    """Check for unsupported scalar operations."""
+    unsupported_scalar_operations = {}  # Map scalars to value to be renamed.
+    for scalar in scalars:
+        if renamings.get(scalar, scalar) != scalar:
+            # Then it either attempts to rename or suppress the scalar.
+            unsupported_scalar_operations[scalar] = renamings.get(scalar)
```

Comment:
Why `.get()` here, and not `renamings[scalar]`?

Reply:
That's odd, last time I tried that I got some sort of error about how the brackets only work for dicts and not for `Mapping` but upon re-running it now I don't see a problem, so I'll change it.